R/method-correlation.R

```diff
@@ -0,0 +1,173 @@
+    df1 <- data1@data
+    recordIdColumn1 <- data1@recordIdColumn
+    naToZero <- data1@imputeZero
+    ancestorIdColumns1 <- data1@ancestorIdColumns
+    allIdColumns1 <- c(recordIdColumn1, ancestorIdColumns1)
+    df2 <- data2@data
+    recordIdColumn2 <- data2@recordIdColumn
+    ancestorIdColumns2 <- data2@ancestorIdColumns
+    allIdColumns2 <- c(recordIdColumn2, ancestorIdColumns2)
+
+
+    ## Initialize and check inputs
+    method <- veupathUtils::matchArg(method)
+    verbose <- veupathUtils::matchArg(verbose)
+
+    # Check that incoming abundance data meets requirements
+    if (!'data.table' %in% class(df1)) {
+      data.table::setDT(df1)
+    }
+
+    # Check that incoming metadata meets requirements
+    if (!'data.table' %in% class(df2)) {
+      data.table::setDT(df2)
+    }
+
+    # Check that variables is the correct class
+    if (!is.null(variables) && !'VariableMetadataList' %in% class(variables)) {
+      stop("The variables argument must be a VariableMetadataList if defined.")
+    }
+
+    computeMessage <- ''
+    veupathUtils::logWithTime(paste("Received df table with", nrow(df1), "samples and", (ncol(df1)-1), "taxa."), verbose)
+
+    # Replace NA values with 0
+    if (naToZero) {
+      veupathUtils::setNaToZero(df1)
+      veupathUtils::logWithTime("Replaced NAs with 0.", verbose)
+    }
+
+    # Ensure the sample IDs, and their order, are the same in both datasets
+    if (!identical(df1[[recordIdColumn1]], df2[[recordIdColumn2]])) {
+      if (!identical(union(df1[[recordIdColumn1]], df2[[recordIdColumn2]]), intersect(df1[[recordIdColumn1]], df2[[recordIdColumn2]]))) {
+        stop("Sample IDs do not match between data1 and data2.")
+      }
+      # Reorder df2 (sample metadata) so that it matches the order in df
+      sampleMetadata <- df2[match(df1[[recordIdColumn1]], df2[[recordIdColumn2]]), ]
+      veupathUtils::logWithTime("Reordered sampleMetadata rows based on abundance data sample id order.", verbose)
+    }
+
+    ## Extract appropriate metadata columns
+    if (!is.null(variables)) {
+      # Use the data shape to find appropriate columns in the metadata
+      inputMetadataCols <- veupathUtils::findColNamesByPredicate(variables, function(x) {identical(x@dataShape@value, "CONTINUOUS")})
+
+      # If any are dates, we'll need to coerce them to numeric for the correlation
+      dateMetadataCols <- veupathUtils::findColNamesByPredicate(variables, function(x) {identical(x@dataShape@value, "CONTINUOUS") && identical(x@dataType@value, "DATE")})
+      if (!!length(dateMetadataCols)) {
+        veupathUtils::logWithTime("Converting date variables to numeric", verbose)
+        df2[, (dateMetadataCols) := lapply(.SD, as.numeric), .SDcols = dateMetadataCols]
+      }
+
+    } else {
+      warning("No variable metadata supplied. Using all numeric columns of sampleMetadata for correlation.")
```

Comment:
i dont like that the behavior is different whether you pass specific variables or not. presumably the default should also include dates, if were going to support them when requested?

Reply:
also will this mean that we treat differently a var w type = number and shape = ordinal depending if variables is present?